pkg/customresourcestate/registry_factory.go

```diff
@@ -217,7 +217,35 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 	switch iter := v.(type) {
 	case map[string]interface{}:
 		for key, it := range iter {
-			ev, err := c.value(it)
+			// Try to deduce `valueFrom`'s value from the current element.
+			var ev *eachValue
+			var err error
+			var didResolveValueFrom bool
+			sValueFrom := c.ValueFrom.String()
+			if !strings.Contains(sValueFrom, ",") &&
```

Comment:
can you explain in a comment what this and the following lines are doing?

Reply:
ACK